File: Python Enrolment System - Basic FIT5136/src/Unit.py
"""
Unit - Represents a unit with a name and credit points.

A unit in the context of the Student Enrollment System typically represents a course or module
that students can enroll in. Each unit has a name and a number of credit points associated with it.

Author: Antoni Erdeg - antoni.erdeg@monash.edu
Version: 1.0
"""

class Unit:
    def __init__(self, code, title, credit_points):
        """
        Constructs a Unit with the specified name and credit points.

        :param name: The name of the unit.
        :param credit_points: The credit points of the unit.
        """
        self.code = code
        self.credit_points = credit_points
        self.title = title

    def get_title(self):
        """
        Returns the title of the unit.

        :return: The title of the unit.
        """
        return self.title

    def get_code(self):
        """
        Returns the name of the unit.

        :return: The name of the unit.
        """
        return self.code

    def get_credit_points(self):
        """
        Returns the credit points of the unit.

        :return: The credit points of the unit.
        """
        return self.credit_points

    def __str__(self):
        """
        Returns a string representation of the unit.

        :return: A string representation of the unit.
        """
        return f"{self.code} - {self.title} ({self.credit_points} credit points)"

File: Java Enrolment System - Basic FIT5136/src/Unit.java
/**
 * Unit - Represents a unit with a name and credit points.
 * <p>
 * A unit in the context of the Student Enrollment System typically represents a course or module
 * that students can enroll in. Each unit has a name and a number of credit points associated with it.
 * </p>
 *
 * <p>
 * @author Antoni Erdeg - antoni.erdeg@monash.edu<br>
 * @version: 1.0
 * </p>
 */
public class Unit {
    // All final variables are constants and should be upper case.
    // Refer to PartTimeStudent for non-constant variable example i.e. courseAdvisor
    private final String CODE;
    private final String TITLE;
    private final int CREDIT_POINTS;

    /**
     * Constructs a Unit with the specified name and credit points.
     *
     * @param code The name of the unit.
     * @param creditPoints The credit points of the unit.
     */
    public Unit(String code, String title, int creditPoints) {
        this.CODE = code;
        this.CREDIT_POINTS = creditPoints;
        this.TITLE = title;
    }

    /**
     * Returns the title of the unit.
     *
     * @return The title of the unit.
     */
    public String getTitle() {
        return TITLE;
    }

    /**
     * Returns the name of the unit.
     *
     * @return The name of the unit.
     */
    public String getCode() {
        return CODE;
    }

    /**
     * Returns the credit points of the unit.
     *
     * @return The credit points of the unit.
     */
    public int getCreditPoints() {
        return CREDIT_POINTS;
    }

    @Override
    public String toString() {
        return CODE + " - "+ TITLE + " (" + CREDIT_POINTS + " credit points)";
    }
}
